### DISCUSSION TITLE
Error when upgrading webpack

### DISCUSSION THREAD

**Discussion by hcxmj** [webpack/webpack#16086] [Q&A]
Hello author,When I upgraded webpack, the following error was prompted, and no relevant solutions were found in the issue.

============
Module not found: Error: Can't resolve 'webpack/hot/log.js' in '/Users/project/web/node_modules/webpack-dev-server/client'
Module not found: Error: Can't resolve 'ansi-html-community' in '/Users/project/web/node_modules/webpack-dev-server/client'
Module not found: Error: Can't resolve 'html-entities' in '/Users/project/web/node_modules/webpack-dev-server/client'
Module not found: Error: Can't resolve 'webpack/hot/emitter.js' in '/Users/project/web/node_modules/webpack-dev-server/client/utils'

 @ ./node_modules/webpack-dev-server/client/index.js?protocol=ws%3A&hostname=localhost&port=3500&pathname=%2Fws&logging=none&reconnect=10
============

Below is my config file and dependency version：

================
const InterpolateHtmlPlugin = require('interpolate-html-plugin');
const HtmlWebpackPlugin = require('html-webpack-plugin');
const CaseSensitivePathsPlugin = require('case-sensitive-paths-webpack-plugin');
const ProgressBarPlugin = require('progress-bar-webpack-plugin');
const HappyPack = require('happypack');
const { CleanWebpackPlugin } = require('clean-webpack-plugin')

const path = require('path');
const webpack = require('webpack');
const autoprefixer = require('autoprefixer');

const getClientEnvironment = require('./env');
const paths = require('../config/paths');
const publicPath = '/';
const publicUrl = '';
const env = getClientEnvironment(publicUrl);
const happyThreadPool = HappyPack.ThreadPool({size: 5});
const protocol = process.env.HTTPS === 'true' ? "https" : "http";
const host = process.env.HOST || 'localhost';

module.exports = {
    entry: [
        // require.resolve('./polyfills'),
        paths.appIndexJs
    ],
    output: {
        path: paths.appBuild,
        pathinfo: true,
        filename: 'static/js/bundle.js',
        publicPath: publicPath
    },
    devtool: 'source-map',
    resolve: {
        fallback: {
            fs: false,
            "events": require.resolve("events/"),
        },
        extensions: ['.js', '.json', '.jsx', ''],
        alias: {
            'react-native': 'react-native-web',
        },
        modules: [path.resolve(__dirname, 'node_modules')],
    },
    module: {
        rules: [
            {
                exclude: [/\.html$/, /\.(js|jsx)(\?.*)?$/, /\.less$/, /\.css$/, /\.json$/, /\.svg$/],
                loader: 'url',
                options: {
                    limit: 10000,
                    name: 'static/media/[name].[hash:8].[ext]',
                }
            },
            {
                test: /\.(js|jsx)$/,
                // use: ['happypack/loader?id=babel'],
                include: paths.appSrc,
                loader: 'babel',
                options: {
                    plugins: [
                        ['import', [{ libraryName: "antd", style: true }]],
                    ],
                    cacheDirectory: true,
                },
                exclude: /node_modules/,
            },
            {
                test: /\.css$/,
                // loader: 'style!css?importLoaders=1!postcss',
                use: [
                    'style-loader',
                    {
                        loader: 'css-loader',
                        options: {
                            importLoaders: 1,
                        }
                    },
                    'postcss-loader'
                ]
            },
            {
                test: /\.json$/,
                loader: 'json',
            },
            {
                test: /\.svg$/,
                loader: 'file',
                options: {
                  name: 'static/media/[name].[hash:8].[ext]'
                }
            },
            {
                test: /\.less$/,
                // loader: 'style!css!postcss!less?{modifyVars:{"@primary-color":"#1DA57A"}}'
                use: [
                    'style-loader',
                    'css-loader',
                    'postcss-loader',
                    {
                        loader: 'less-loader',
                        options: {
                            modifyVars: {
                                "@primary-color":"#1DA57A"
                            }
                        }
                    }
                ]
            },
        ],
    },
    plugins: [
        new HtmlWebpackPlugin({
            inject: true,
            template: paths.appHtml,
        }),
        new InterpolateHtmlPlugin(env.raw),
        new CleanWebpackPlugin(),
        new CaseSensitivePathsPlugin(),
        new webpack.DefinePlugin({
            'process.env.NODE_ENV': JSON.stringify(process.env.NODE_ENV || 'development'),
            __DEV__: process.env.NODE_ENV === 'production' || true,
        }),
        new ProgressBarPlugin(),
        new HappyPack({
            id: 'babel',
            loaders: ['babel-loader?cacheDirectory'],
            threadPool: happyThreadPool,
        }),
        new HappyPack({
            id: 'css',
            loaders: ['css-loader',],
            threadPool: happyThreadPool
        }),
        new webpack.LoaderOptionsPlugin({
            options: {
              context: __dirname,
              postcss: [
                autoprefixer
              ]
            }
        })        
    ],
    devServer: {
        compress: false,
        hot: true,
        https: protocol === "https",
        host: host,
        port: 3500,
        proxy: {
            '/sso': 'http://apm.qiyi.domain',
            '/apmconfig': 'http://apm.qiyi.domain',
            '/wsgi': 'http://apm.qiyi.domain',
            '/rest': 'http://apm.qiyi.domain',
            '/druid': 'http://apm.qiyi.domain',
            '/kylin': 'http://apm.qiyi.domain',
            '/apm': 'http://apm.qiyi.domain',
            '/qmas': 'http://qmasapi.qiyi.domain',
            '/api/GetPassword': 'http://itpwd.qiyi.domain',
            '/urls/yourls-api.php': 'http://arch.qiyi.domain',
            '/api/open': 'http://hubble.qiyi.domain/',
            '/api/alarm': 'http://hubble.qiyi.domain/',
            // '/alarm': 'http://hubble-pre.qiyi.domain'
        },
        historyApiFallback: {
            disableDotRule: true,
        },
        static: {
            directory: paths.staticPath,
            publicPath: paths.appPublic,
            watch: true,
        },
        client: {
            logging: 'none',
        }
    }
}

====================

    "webpack": "^5.74.0",
    "webpack-cli": "^4.10.0",
    "webpack-dev-server": "^4.9.3",
    "html-webpack-plugin": "^5.5.0",

 "scripts": {
    "start": "cross-env NODE_ENV=development webpack serve --config config/webpack.dev.js",
  },

node version: 14.15.3

**Comment by alexander-akait** [comment]
Sounds like you have corrupted node_modules or change resolution logic using package.json (also check webpack-dev-server version)

**Reply by hcxmj** [reply]
I checked the node_module package, and it seems that there should be a problem with configuration parsing. The webpack file that webpack-dev-server depends on is not obtained.
I recheck to see the parsing path.Thank you @alexander-akait
